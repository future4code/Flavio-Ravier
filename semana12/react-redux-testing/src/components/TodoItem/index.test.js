import React from 'react'
import {shallow} from 'enzyme'
import {DeleteButton, TodoItem, TodoListItem} from './index'
import renderer from "react-test-renderer";

describe('Testa componente TodoItem', () => {
	test('Deve chamar a funcao onDeleteTodo quando clicar no botao', () => {
		// Preparacao
		const onDeleteTodoMock = jest.fn()
		const component = shallow(<TodoItem onCompleteTodo={jest.fn()} onDeleteTodo={onDeleteTodoMock} todo={{}}/>)

		const deleteButton = component.find(DeleteButton)

		// Execucao
		deleteButton.simulate('click')

		// Verficacao
		expect(onDeleteTodoMock).toHaveBeenCalledTimes(1)
	})

	test('Deve passar props corretamente para um TodoListItem', () => {
		// Preparacao
		const mockTodo = {
			id: '1',
			text: 'bla',
			completed: true
		}

		// Execucacao
		const component = shallow(<TodoItem onCompleteTodo={jest.fn()} onDeleteTodo={jest.fn()} todo={mockTodo}/>)

		// Verificacao
		const todoListItem = component.find(TodoListItem)

		expect(todoListItem.props().completed).toBe(true)
		expect(todoListItem.props().children).toBe('bla')
	})

	test('Snapshot', () => {
		const tree = renderer.create(<TodoItem todo={{}}/>).toJSON();
		expect(tree).toMatchSnapshot();
	})
})
