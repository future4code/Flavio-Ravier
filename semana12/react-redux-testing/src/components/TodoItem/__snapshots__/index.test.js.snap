// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testa componente TodoItem Snapshot 1`] = `
<div
  className="sc-bdVaJa gLJfMc"
>
  <li
    className="sc-bwzfXH cozqzq"
    onClick={[Function]}
  />
  <span
    className="sc-htpNat eAAkmT"
    onClick={[Function]}
  >
    X
  </span>
</div>
`;
